internal/meta/annotation.go

```diff
@@ -19,55 +15,13 @@ var (
 	xCreatorAnnotation        = project.Name() + ".x-giantswarm.io/creator"
 	xInstallationAnnotation   = project.Name() + ".x-giantswarm.io/installation"
 	xObjectHashAnnotation     = project.Name() + ".x-giantswarm.io/object-hash"
-	xPreviousConfigAnnotation = project.Name() + ".x-giantswarm.io/previous-config"
 	xProjectVersionAnnotation = project.Name() + ".x-giantswarm.io/project-version"
 )
 
 type ConfigVersion struct{}
 
 func (ConfigVersion) Key() string { return configVersionAnnotation }
 
-type XPreviousConfig struct{}
-
-func (XPreviousConfig) Key() string { return xPreviousConfigAnnotation }
-
-func (XPreviousConfig) Get(o Object) (v1alpha1.ConfigStatusConfig, error) {
```

Comment:
config CRs are deprecated and the annotation is not used. So we can drop the apiextensions dependency.